### ISSUE TITLE
[Bug] Gradle build fails after upgrading to Firebase 12.0.0

### ISSUE THREAD

**Issue by bartutiryakioglu** [firebase/firebase-unity-sdk#1035]
### Description

After upgrading Firebase to 12.0.0, I had to upgrade my target SDK version to 34. I can't build for android with this change. Gradle build fails with **launcher:processReleaseResources** task and with the error **AAPT2 aapt2-7.0.3-7396180-osx Daemon #0: Unexpected error during link, attempting to stop daemon.**. I've also Facebook, Applovin and Appsflyer in my repository. But also all of them upgraded to their latest version. What causes that error?

### Firebase Unity SDK Version

12.0.0

### Unity editor version

2021.3.29f1

### Installation Method

.unitypackage

### Additional SDKs you are using

Facebook, Applovin, Appsflyer

### Targeted Platform(s)

Android

### Unity editor platform

Mac

### Scripting Runtime

IL2CPP

### Release Distribution Type

Pre-built SDK from https://firebase.google.com/download/unity

### If using CocoaPods for Apple platforms, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

👀 Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by argzdev** [comment]
Hey @bartutiryakioglu, thanks for reaching out. I'm trying to build our quickstart app for Firebase Analytics, and it seems to work alright. Could you specify which particular product are you using? Also have you tried using the `Asset > External Dependency Manager > Android Resolver > Resolve or Force Resolve` option and see if that makes any difference?

**Comment by jrnavarro0213** [comment]
I'm having the same issue. Build issues with firebase, keeps giving me a jettify issue.

**Comment by argzdev** [comment]
Hey @jrnavarro0213, could you share what Firebase product are you using?

**Comment by S-Chanchon11** [comment]
I'm also having similar issue. But my error message is "Could not find com.google.firebase:firebase-app-unity:12.0.0."

**Comment by xiong-s** [comment]
I don't think this issue is related to Firebase.
When you set the Target Android SDK to 34, you should use BuildTools 34. However, the Android SDK included with Unity 2021.3 contains BuildTools 30. (`Unity Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\30.0.2`)
I resolved the OP's issue by adding a line `android.aapt2FromMavenOverride=/path/to/BuildTool34/aapt2` to `Plugins/Android/gradleTemplate.properties`.
(If you are on a Mac, don't forget to run `chmod +x aapt2`)

But now, I have encountered a new problem...
I think I should submit a new issue for this new problem. I am currently trying to create an MCVE (Minimal, Complete, and Verifiable Example) for the issue.

**Comment by gulliversgames** [comment]
@xiong-s can you please reference that issue here as well when you open?

thanks in advance!
